### DISCUSSION TITLE
How to generate routes from i18n folder?

### DISCUSSION THREAD

**Discussion by juzhiyuan** [facebook/docusaurus#5682] [Q&A]
Hi, I have one case here:

There has one post called `post-b.md` in Chinese only, and one post is called `post-a.md` in English only, so I put files like:

```
/website/blog/post-a.md

/website/i18n/zh/docusaurus-plugin-content-blog/post-b.md
```

**Expected Result**
- When I visit the blog in English, it shows `post-a` only;
- When I visit the blog in Chinese, it shows `post-b` only.

**Actually Result**
Due to `post-b.md` is in `i18n` directory only, so it only shows `post-a` in English and Chinese mode.

**Comment by juzhiyuan** [comment]
By reviewing source codes, Docusaurus will generate a Routes.js file 🤔

**Comment by juzhiyuan** [comment]
Do I have to have posts in multiple languages at the same time? Could we merge posts from multiple langs?

**Reply by slorber** [reply]
> There has one post called post-b.md in Chinese only, and one post is called post-a.md in English only, so I put files like:

> Do I have to have posts in multiple languages at the same time?

Docusaurus uses a fallback/override strategy and expects the content to exist in the default language, so technically you cannot create a blog post only for Chinese without existing it in English in the first place.

This is a current limitation that I am aware of, not sure how to solve it, but there are possible workarounds:
- As @Josh-Cena suggested: create both files, and use the `draft` frontmatter on the blog post you want to hide
- You can also create one config per locale and build each locale independently with `docusaurus build --config docusaurus.config.zh.js`: this gives you the opportunity to escape this i18n fallback mechanism, as if your site was only supporting a single locale, and you could create folders like `blog_en` and `blog_zh`

**Reply by juzhiyuan** [reply]
ok, cc @bzp2010 and @baoyuantop to know this tip :)

**Comment by Josh-Cena** [comment]
A hack I tell people on Discord to use is to make both post-a and post-b available in both folders, and put draft: true front matter on pages you don't want to show in one particular locale. This is kinda hacky and I also wish Docusaurus can stop assuming that translated sites only change their contents but not their site structure, layout, etc.🧐

**Reply by slorber** [reply]
Yes I agree we should find a way to configure a site differently per locale. 

I thought about using locale customizations callbacks in `docusaurus.config.js` but it does not look very clean

Using `docusaurus build --config docusaurus.config.zh.js` is a possibility but we don't document it much (+ tips on how to share common configs), and you'd have to build each locale independently in the correct order (locale at `/` first).

Maybe having first-class support for `docusaurus.config.<locale>.js` could make sense, and `docusaurus build` would handle it by default?

**Reply by Josh-Cena** [reply]
I created a feature request yesterday on Canny because this one is slightly non-trivial: https://docusaurus.io/feature-requests/p/exclude-certain-pages-in-certain-locales

Basically I want it to be realized at the plugin level, such that each plugin can do context-specific content loading. Not sure how much work there would be at tweaking the core architecture

**Reply by juzhiyuan** [reply]
> Basically I want it to be realized at the plugin level

Agree +1

**Reply by juzhiyuan** [reply]
And for now, we will try to translate the posts both in English and Chinese 😄
